### PR TITLE
ADCM-3119 Add ability to rename cluster

### PR DESCRIPTION
**Pull request by DanBalalan**

### CODE REVIEW

File: python/api/cluster/views.py [hunk 1]
```diff
@@ -125,6 +128,15 @@ def patch(self, request, *args, **kwargs):
         serializer = self.get_serializer(obj, data=request.data, partial=True)
         return update(serializer)
 
+    @audit
+    def put(self, request, *args, **kwargs):
+        """
```
Comment (a-alferov) [inline python/api/cluster/views.py:133]: It's a useless doc-strings

File: python/cm/tests/test_cluster.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            "Use-forbidden&chars",
+            "Use-forbidden?chars",
+            "Use-forbidden!chars",
+        ]
+
+        for name in valid_names:
+            response = self.client.patch(
+                path=url, data={"name": name}, content_type=APPLICATION_JSON
+            )
+            assert response.status_code == status.HTTP_200_OK
```
Comment (a-alferov) [inline python/cm/tests/test_cluster.py:48]: Please use built-in asserts, for example `self.assertEqual()`

**Review by a-alferov — CHANGES_REQUESTED** [review]

File: python/api/validators.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from rest_framework.serializers import ValidationError
+
+CLUSTER_NAME_PATTERN = re.compile(
+    r"^[a-zA-Z]"  # starts with latin letter (upper/lower case)
+    r"[a-zA-Z0-9\-\. ]*?"  # latin letters (upper/lower case), digits, hyphens, dots, whitespaces
+    r"[a-zA-Z0-9]$"  # ends with latin letter (upper/lower case) or digit
+)  # as a result of this pattern min_length = 2
+
+
+class CharFieldMatchValidator:
```
Comment (a-alferov) [inline python/api/validators.py:12]: Use pls `RegexValidator` for parent class, for example https://github.com/arenadata/adcm/pull/2015/files#diff-affe7e9dda5e6a57a496094f26a4c70ae4f1f7603a14b5c3f1a6199221ce19c1R48

**Review by a-alferov — CHANGES_REQUESTED** [review]

File: python/cm/tests/test_cluster.py (hunk 2 above)
Comment (DanBalalan) [reply python/cm/tests/test_cluster.py]: done

File: python/api/cluster/views.py (hunk 1 above)
Comment (DanBalalan) [reply python/api/cluster/views.py]: removed

File: python/api/validators.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, regexp: re.Pattern):
+        self.regexp = regexp
+
+    def __call__(self, value):
+        if not self.regexp.match(value):
+            raise ValidationError(f"`{value}` does not match `{self.regexp.pattern}` pattern")
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply python/api/validators.py]: done

File: python/api/validators.py (hunk 3 above)
Comment (DanBalalan) [reply python/api/validators.py]: done

**Review by a-alferov — APPROVED** [review]

File: python/api/validators.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import re
+
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.core.validators import RegexValidator
+
+from cm.errors import AdcmEx
+
+CLUSTER_NAME_PATTERN = re.compile(
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply python/api/validators.py]: moved to settings; refactored

File: python/adcm/tests/base.py [hunk 6]
```diff
@@ -60,10 +60,15 @@ def setUp(self) -> None:
             self.test_bundle_filename,
         )
 
-    def login(self):
+    def login(self, username=None, password=None):
+        if not username:
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply python/adcm/tests/base.py]: refactored

File: python/adcm/tests/base.py [hunk 7]
```diff
@@ -60,10 +60,15 @@ def setUp(self) -> None:
             self.test_bundle_filename,
         )
 
-    def login(self):
+    def login(self, username=None, password=None):
```
[the remark this answers is not in the document]
Comment (DanBalalan) [reply python/adcm/tests/base.py]: refactored

**Review by winmasta — APPROVED** [review]

**Pull request merged**